libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1129,6 +1135,91 @@ map<string, Json::Value> StandardCompiler::parseAstFromInput(StringMap const& _s
 	return sourceJsons;
 }
 
+Json::Value StandardCompiler::importEvmAssembly(StandardCompiler::InputsAndSettings _inputsAndSettings)
+{
+	solAssert(_inputsAndSettings.language == "EvmAssemblyJSON");
+	solAssert(1 == _inputsAndSettings.sources.size());
+
+	bool const binariesRequested = isBinaryRequested(_inputsAndSettings.outputSelection);
+	bool const wildcardMatchesExperimental = false;
+	Json::Value output = Json::objectValue;
+	if (binariesRequested)
+	{
+		evmasm::EVMAssemblyStack stack(_inputsAndSettings.evmVersion);
+		std::string const& sourceName = _inputsAndSettings.sources.begin()->first; // result of structured binding can only be used within lambda from C++20 on.
+		std::string const& source = _inputsAndSettings.sources.begin()->second;
+		stack.parseAndAnalyze(sourceName, source);
+		stack.assemble();
+		if (stack.compilationSuccessful())
+		{
+			// EVM
+			Json::Value evmData(Json::objectValue);
+			if (isArtifactRequested(_inputsAndSettings.outputSelection, sourceName, "", "evm.assembly", wildcardMatchesExperimental))
+				evmData["assembly"] = stack.assemblyString(sourceName, {{sourceName, source}});
+			if (isArtifactRequested(_inputsAndSettings.outputSelection, sourceName, "", "evm.legacyAssembly", wildcardMatchesExperimental))
+				evmData["legacyAssembly"] = stack.assemblyJSON(sourceName);
+
+			if (isArtifactRequested(
+					_inputsAndSettings.outputSelection,
+					sourceName,
+					"",
+					evmObjectComponents("bytecode"),
+					wildcardMatchesExperimental
+					))
+			{
+				evmData["bytecode"] = collectEVMObject(
+					_inputsAndSettings.evmVersion,
+					stack.object(sourceName),
+					stack.sourceMapping(sourceName),
+					{},
+					false,
+					[&](string const& _element)
+					{
+						return isArtifactRequested(
+							_inputsAndSettings.outputSelection,
+							sourceName,
+							"",
+							"evm.bytecode." + _element,
+							wildcardMatchesExperimental);
+					});
+			}
+
+			if (isArtifactRequested(
+					_inputsAndSettings.outputSelection,
+					sourceName,
+					"",
+					evmObjectComponents("deployedBytecode"),
+					wildcardMatchesExperimental
+					))
+				evmData["deployedBytecode"] = collectEVMObject(
+					_inputsAndSettings.evmVersion,
+					stack.runtimeObject(sourceName),
+					stack.runtimeSourceMapping(sourceName),
+					{},
+					true,
+					[&](string const& _element)
+					{
+						return isArtifactRequested(
+							_inputsAndSettings.outputSelection,
+							sourceName,
+							"",
+							"evm.deployedBytecode." + _element,
+							wildcardMatchesExperimental);
+					});
+
+			Json::Value contractData(Json::objectValue);
+			if (!evmData.empty())
+				contractData["evm"] = evmData;
+
+			Json::Value contractsOutput = Json::objectValue;
+			contractsOutput[sourceName][""] = contractData;
+			if (!contractsOutput.empty())
```

Comment:
Can `contractsOutput` every be empty if we assign to it in the line above?

Reply:
True!